src/Microsoft.Health.Dicom.Functions/BlobMigration/CleanupDeletedDurableFunction.Orchestration.cs

```diff
@@ -0,0 +1,110 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Dicom.Core.Features.ChangeFeed;
+using Microsoft.Health.Dicom.Core.Features.Model;
+using Microsoft.Health.Dicom.Functions.BlobMigration.Models;
+using Microsoft.Health.Dicom.Functions.Migration;
+using Microsoft.Health.Operations.Functions.DurableTask;
+
+namespace Microsoft.Health.Dicom.Functions.BlobMigration;
+
+public partial class CleanupDeletedDurableFunction
+{
+    /// <summary>
+    /// Asynchronously deletes dangling DICOM deleted instances.
+    /// It goes through DICOM instances in the past, deletes old format blob.
+    /// </summary>
+    /// <param name="context">The context for the orchestration instance.</param>
+    /// <param name="logger">A diagnostic logger.</param>
+    /// <returns>A task representing the <see cref="CleanupDeletedFilesAsync"/> operation.</returns>
+    [FunctionName(nameof(CleanupDeletedFilesAsync))]
+    public async Task CleanupDeletedFilesAsync(
+        [OrchestrationTrigger] IDurableOrchestrationContext context,
+        ILogger logger)
+    {
+        // The ID should be a GUID as generated by the trigger, but we'll assert here just to make sure!
+        EnsureArg.IsNotNull(context, nameof(context)).ThrowIfInvalidOperationId();
+        BlobMigrationCheckpoint input = context.GetInput<BlobMigrationCheckpoint>();
+        EnsureArg.IsNotNull(input, nameof(input));
+        EnsureArg.IsNotNull(input.Batching, nameof(input.Batching));
+
+        logger = context.CreateReplaySafeLogger(logger);
+
+        // Get the max watermark of all deleted instances for the exit scenario
+        long maxWatermark = input.MaxWatermark ?? await context.CallActivityWithRetryAsync<long>(
+                 nameof(GetMaxDeletedChangeFeedWatermarkAsync),
+                 _options.RetryOptions,
+                 new CleanupDeletedBatchArguments
```

Comment:
You can simply pass in a nullable timestamp instead if you'd like and then read in the input as `DateTime?`

Reply:
Yes